modules/tests/src/smithy4s/tests/PizzaClientSpec.scala

```diff
@@ -111,29 +113,49 @@ abstract class PizzaClientSpec extends IOSuite {
     GenericClientError("generic error message for 418")
   )
 
-  clientTestForError(
-    "Handle error w/o a discriminator header nor a unique status code",
-    Response(status = Status.ProxyAuthenticationRequired)
-      .withEntity(
-        Json.obj("message" -> Json.fromString("generic client error message"))
-      ),
-    unknownResponse(
-      407,
-      Map("Content-Length" -> "42", "Content-Type" -> "application/json"),
-      """{"message":"generic client error message"}"""
-    )
+  clientAssertError[RawErrorResponse](
+    "Handle error with a discriminator but can't be decoded",
+    Response[IO](status = Status.NotFound)
+      .withEntity("malformed body")
+      .withHeaders(Header.Raw(CIString("X-Error-Type"), "NotFoundError")),
+    error => {
+      expect(error.code == 404) &&
+      expect(
+        error.headers == Map(
+          CaseInsensitive("X-Error-Type") -> List("NotFoundError")
+        )
+      ) &&
+      expect(error.body == "malformed body") &&
+      expect(
+        error.failedDecodeAttempt.discriminator == HttpDiscriminator.NameOnly(
+          "NotFoundError"
+        )
+      ) &&
+      expect(error.failedDecodeAttempt.isInstanceOf[DecodingFailure])
+    }
   )
 
-  private def unknownResponse(
-      code: Int,
-      headers: Map[String, String],
-      body: String
-  ): UnknownErrorResponse =
-    UnknownErrorResponse(
```

Comment:
**issue:** `UnknownErrorResponse` is no longer used, we should remove it.

In addition, the docs in `modules/docs/markdown/03-protocols/04-simple-rest-json/03-client.md` need updating because they still refer to `UnknownErrorResponse` - they should now mention `RawErrorResponse`, but there might be other differences too, e.g. we could mention when a particular `FailedDecodeAttempt` shows up

Reply:
and finally, we'll need a CHANGELOG entry for this change 😅

Reply:
ok, dealt with those